From 3d498f9886c7ae5002764b4bf4a400331f03de02 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 11 Aug 2026 08:47:40 +0000
Subject: [PATCH] Relax requirement for visible flag to sdcards

The vast majority of sdcard readers are stable enough to be declared by the API. (I see no counter-example)
FBE broke adoptable storage with SDCard, hence this need.

Change-Id: Ia616671c03562d1eadaff5531a5c708a62d7ad3a

[Port to A16] A16 uses vol.getDisk()/setMountFlags accessors; condition widened with isSd() as in original.

Change-Id: I7ff5808567d145937345c6474c308cd74997443b
---
 .../core/java/com/android/server/StorageManagerService.java    | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/StorageManagerService.java b/services/core/java/com/android/server/StorageManagerService.java
index ebc1eef1cc99..deadeecd522a 100644
--- a/services/core/java/com/android/server/StorageManagerService.java
+++ b/services/core/java/com/android/server/StorageManagerService.java
@@ -1667,7 +1667,8 @@ class StorageManagerService extends IStorageManager.Stub
 
             // Adoptable public disks are visible to apps, since they meet
             // public API requirement of being in a stable location.
-            if (vol.getDisk().isAdoptable()) {
+            // Assume all SDs match this as well
+            if (vol.getDisk().isAdoptable() || vol.getDisk().isSd()) {
                 vol.setMountFlags(vol.getMountFlags() | VolumeInfo.MOUNT_FLAG_VISIBLE_FOR_WRITE);
             } else if (vol.getDisk().isSd()) {
                 vol.setMountFlags(vol.getMountFlags() | VolumeInfo.MOUNT_FLAG_VISIBLE_FOR_WRITE);
-- 
2.43.0

